specificatie/zoek-personen.yaml

```diff
@@ -0,0 +1,333 @@
+              example: "de"
+            voornamen:
+              description: |
+                De verzameling namen die, gescheiden door spaties, aan de geslachtsnaam voorafgaat. ** Bij deze query-parameter is het gebruik van een [wildcard](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.3.0/features/wildcard.feature) toegestaan in combinatie met minimaal 2 karakters.** **Zoeken met tekstvelden is [case-Insensitive](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/case_insensitive.feature).**
+              type: string
+              pattern: ^(\?+|\*)?[a-zA-Z0-9À-ÿ' '\-\']{1,200}(\?+|\*)?$
+              example: "Dirk"
+    ZoekMetNaamEnGemeenteVanInschrijving:
+      required:
+        - geslachtsnaam
+        - gemeenteVanInschrijving
+        - voornamen
+      allOf:
+        - $ref: '#/components/schemas/PersonenQuery'
+        - type: object
+          properties:
+            inclusiefOverledenPersonen:
+              $ref: '#/components/schemas/InclusiefOverledenPersonen'
+            geslachtsaanduiding:
+              description: |
+                Geeft aan dat de persoon een man of een vrouw is, of dat het geslacht (nog) onbekend is.
+              type: string
+              pattern: ^([Mm]|[Vv]|[Oo])$
+              example: "V"
+            geslachtsnaam:
+              description: |
+                De (geslachts)naam waarvan de eventueel aanwezige voorvoegsels zijn afgesplitst. **Gebruik van de wildcard is toegestaan. Zie [feature-beschrijving](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.3.0/features/wildcard.feature)** **Zoeken met tekstvelden is [case-Insensitive](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/case_insensitive.feature).**
+              type: string
+              pattern: ^(\?+|\*)?[a-zA-Z0-9À-ÿ '\-\']{1,200}(\?+|\*)?$
+              example: "Vries"
+            voorvoegsel:
+              description: |
+                Deel van de geslachtsnaam dat vooraf gaat aan de rest van de geslachtsnaam. Het zoeken op het voorvoegsel is [case-Insensitive](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/case_insensitive.feature).**
+              type: string
+              pattern: ^[a-zA-Z0-9À-ÿ' '\-\']{1,10}$
+              example: "de"
+            voornamen:
+              description: |
+                De verzameling namen die, gescheiden door spaties, aan de geslachtsnaam voorafgaat. ** Bij deze query-parameter is het gebruik van een [wildcard](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.3.0/features/wildcard.feature) toegestaan in combinatie met minimaal 2 karakters.** **Zoeken met tekstvelden is [case-Insensitive](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/case_insensitive.feature).**
+              type: string
+              pattern: ^(\?+|\*)?[a-zA-Z0-9À-ÿ' '\-\']{1,200}(\?+|\*)?$
```

Comment:
ook hier To Do:
- niet starten met wildcard
- wanneer wel wildcard gebruikt is, dan minimaal 3 letters

Reply:
> ook hier To Do:
> 
>     * niet starten met wildcard
> 
>     * wanneer wel wildcard gebruikt is, dan minimaal 3 letters



> ook hier To Do:
> 
>     * niet starten met wildcard
> 
>     * wanneer wel wildcard gebruikt is, dan minimaal 3 letters

Is het nodig om te beperken tot minimaal 3 letters bij gebruik van wildcard? Voornamen filteren met * is hetzelfde als geen filter op voornamen en alleen ? in de voornamen zoekstring betekent dat je alleen maar personen wilt vinden waarvan de voornaam uit 1 letter bestaat. En het zoekresultaat wordt al beperkt door geslachtsnaam en geboortedatum

Reply:
bij gebruik van voornamen in combinatie met geslachtsnaam+geboortedatum is het denk ik heel redelijk om geen extra eisen aan voornamen te zetten. Bij zoeken met gemeenteVanInschrijving+geslachtsnaam+voornamen is voornamen een verplichte parameter en moeten daar denk ik wel eisen gesteld worden. Wanneer dan voornamen=* toegestaan zou worden, heb je het verplicht maken van parameter voornamen in feite opgeheven.